pkg/virt-launcher/virtwrap/network/cloud-init-network_test.go

```diff
@@ -0,0 +1,225 @@
+						Expect(intSubnet.Routes[routeNum].Netmask).To(Equal("255.255.255.0"))
+						if routeNum == intNum {
+							Expect(intSubnet.Routes[routeNum].Gateway).To(Equal(""))
+						} else {
+							Expect(intSubnet.Routes[routeNum].Gateway).To(Equal("10." + intString + ".0.1"))
+						}
+						routeNum++
+					}
+				}
+
+				Expect(intConfig.Address).To(BeNil())
+				Expect(intConfig.Search).To(BeNil())
+				Expect(intConfig.Destination).To(Equal(""))
+				Expect(intConfig.Gateway).To(Equal(""))
+				Expect(intConfig.Metric).To(Equal(0))
+
+				intNum++
+			}
+
+		})
+	})
+})
+
+func buildMockSriovNetwork(mockNetwork *MockNetworkHandler, count int) {
+	netInts := make(map[string]*netlink.Dummy, count)
+	intNum := 1
+	for intNum <= count {
+
+		var addrList []netlink.Addr
+		var routeList []netlink.Route
+		var ipAddress netlink.Addr
+		var macAddress net.HardwareAddr
+
+		intName := "net" + strconv.Itoa(intNum)
+		netInts[intName] = &netlink.Dummy{
+			LinkAttrs: netlink.LinkAttrs{
+				Index: count,
+				MTU:   1400 + intNum,
+				Name:  intName,
+				Alias: intName,
+			},
+		}
+
+		address := &net.IPNet{IP: net.IPv4(10, byte(intNum), 0, 2), Mask: net.CIDRMask(24, 32)}
+		ipAddress = netlink.Addr{IPNet: address}
+		addrList = []netlink.Addr{ipAddress}
+		macAddrStr := "de:ad:be:af:00:0" + strconv.Itoa(intNum)
+		macAddress, _ = net.ParseMAC(macAddrStr)
+
+		routeList = append(routeList, netlink.Route{Src: net.IPv4(10, byte(intNum), 0, 2)})
+
+		if intNum == 1 {
+			gw := net.IPv4(10, byte(intNum), 0, 1)
+			routeList = append(routeList, netlink.Route{Gw: gw})
+		} else if intNum > 1 {
+			routeCount := 1
+			for routeCount <= intNum {
+				staticRoute := netlink.Route{
+					Dst: &net.IPNet{IP: net.IPv4(10, byte(intNum), byte(routeCount), 0), Mask: net.CIDRMask(24, 32)},
+					Gw:  net.IPv4(10, byte(intNum), 0, 1),
+				}
+				routeList = append(routeList, staticRoute)
+				routeCount++
+			}
+			gwRoute := netlink.Route{
+				Dst: &net.IPNet{IP: net.IPv4(10, byte(intNum), byte(routeCount), 0), Mask: net.CIDRMask(24, 32)},
+			}
+
+			routeList = append(routeList, gwRoute)
+		}
+
+		netlink.AddrAdd(netInts[intName], &ipAddress)
+
+		mockNetwork.EXPECT().LinkByName(intName).Return(netInts[intName], nil)
+		mockNetwork.EXPECT().AddrList(netInts[intName], netlink.FAMILY_V4).Return(addrList, nil)
+		mockNetwork.EXPECT().RouteList(netInts[intName], netlink.FAMILY_V4).Return(routeList, nil)
+		mockNetwork.EXPECT().GetMacDetails(intName).Return(macAddress, nil)
+
+		intNum++
+	}
+}
+
+func newSriovVMI(namespace, name string, count int) *v1.VirtualMachineInstance {
+	vmi := &v1.VirtualMachineInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1.VirtualMachineInstanceSpec{
+			Domain:   v1.NewMinimalDomainSpec(),
+			Networks: []v1.Network{*v1.DefaultPodNetwork()},
+		},
+	}
+	intNum := 1
+	for intNum <= count {
+		intName := "sriov" + strconv.Itoa(intNum)
+		sriovNetwork := v1.Network{
```

Comment:
While this may work as of now, shouldn't we perhaps also add interfaces to reflect VMI networks defined in the spec for completeness?

Reply:
It should also get a default pod network iirc. We can add other types (I had quite a bit of trouble initially getting any testing going as I'm not much of a dev :P)